nucypher/blockchain/eth/registry.py

```diff
@@ -427,3 +449,38 @@ def read(self) -> list:
             else:
                 raise
         return registry_data
+
+
+class IndividualAllocationRegistry(InMemoryAllocationRegistry):
+
+    REGISTRY_NAME = "individual_allocation_ABI.json"
+
+    def __init__(self, beneficiary_address: str, contract_address: str, contract_abi=None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.beneficiary_address = beneficiary_address
+        self.contract_address = contract_address
+
+        if not contract_abi:
+            # Download individual allocation template to extract contract_abi
+            individual_allocation_template = json.loads(self.fetch_latest_publication())
+            if len(individual_allocation_template) != 1:
+                raise self.IllegalRegistry("Individual allocation template must contain a single entry")
```

Comment:
Minor nitpick: I would vote for `InvalidRegistry` instead of `IllegalRegistry`.

Reply:
✅